nomos-services/blend/src/core/tests/mod.rs

```diff
@@ -0,0 +1,181 @@
+mod utils;
+
+use std::time::Duration;
+
+use groth16::Field as _;
+use nomos_core::crypto::ZkHash;
+use nomos_time::SlotTick;
+use poq::CORE_MERKLE_TREE_HEIGHT;
+use tokio::time::{sleep, timeout};
+
+use crate::{
+    core::{
+        initialize, post_initialize,
+        processor::tests::MockCoreAndLeaderProofsGenerator,
+        run_event_loop,
+        tests::utils::{
+            NodeId, TestBlendBackend, TestNetworkAdapter, dummy_overwatch_resources,
+            new_membership, new_stream, settings,
+        },
+    },
+    epoch_info::EpochHandler,
+    membership::{MembershipInfo, ZkInfo},
+    settings::TimingSettings,
+    test_utils::{
+        crypto::MockProofsVerifier,
+        epoch::{OncePolStreamProvider, TestChainService},
+    },
+};
+
+type RuntimeServiceId = ();
+
+/// Check if the service keeps running after it receives a new session where
+/// it's still core. Also, check if it stops after the session transition period
+/// if it receives another new session that doesn't meet the core node
+/// conditions.
+#[test_log::test(tokio::test)]
+async fn complete_old_session_after_main_loop_done() {
+    let minimal_network_size = 2;
+    let (membership, local_private_key) = new_membership(minimal_network_size);
+
+    // Create settings.
+    let (settings, _recovery_file) = settings(
```

Comment:
```suggestion
    let (settings, _) = settings(
```

Reply:
If we use `_`, the `recovery_file: NamedTempFile` instance is dropped and the corresponding temp file is deleted from disk. Although this test currently doesn't produce any state to be stored, but I think it's better to not drop the file instance for the future.